smack-experimental/src/main/java/org/jivesoftware/smackx/file_metadata/element/FileMetadataElement.java

```diff
@@ -0,0 +1,262 @@
+/**
+ *
+ * Copyright 2020 Paul Schaub
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.file_metadata.element;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.jivesoftware.smack.packet.Element;
+import org.jivesoftware.smack.packet.ExtensionElement;
+import org.jivesoftware.smack.packet.NamedElement;
+import org.jivesoftware.smack.packet.XmlEnvironment;
+import org.jivesoftware.smack.util.EqualsUtil;
+import org.jivesoftware.smack.util.HashCode;
+import org.jivesoftware.smack.util.Objects;
+import org.jivesoftware.smack.util.XmlStringBuilder;
+import org.jivesoftware.smackx.file_metadata.element.child.DateElement;
+import org.jivesoftware.smackx.file_metadata.element.child.DescElement;
+import org.jivesoftware.smackx.file_metadata.element.child.DimensionsElement;
+import org.jivesoftware.smackx.file_metadata.element.child.LengthElement;
+import org.jivesoftware.smackx.file_metadata.element.child.MediaTypeElement;
+import org.jivesoftware.smackx.file_metadata.element.child.NameElement;
+import org.jivesoftware.smackx.file_metadata.element.child.SizeElement;
+import org.jivesoftware.smackx.hashes.HashManager;
+import org.jivesoftware.smackx.hashes.element.HashElement;
+
+/**
+ * File metadata element as defined in XEP-0446: File Metadata Element.
+ *
+ * This element is used in a generic way to provide information about files, eg. during file sharing.
+ */
+public final class FileMetadataElement implements ExtensionElement {
+
+    public static final String ELEMENT = "file";
+    public static final String NAMESPACE = "urn:xmpp:file:metadata:0";
+
+    private final Set<NamedElement> children = new HashSet<>();
```

Comment:
It is very uncommon that every child element gets its own class in such a case. XEP-0446 § 3 has a nice description of all child elements, which are all optional and, as far as I can tell, can only appear at most once. So why not simply create 8 member fields for those in `FileMetadataElement` (note that the situation may be different for `<thumbnail/>`? I think this would make the code much more readable, accessible and maintainable.

Reply:
Yeah I wasn't sure about the cardinality of the child elements. HashElement for example may occur multiple times.
But for name etc. I can definitely see your point.